vortex-array/src/vtable/canonical.rs

```diff
@@ -1,12 +1,20 @@
 use vortex_error::{VortexError, VortexExpect, VortexResult};
 
+use crate::builders::ArrayBuilder;
 use crate::encoding::Encoding;
 use crate::{Array, Canonical};
 
 /// Encoding VTable for canonicalizing an array.
 #[allow(clippy::wrong_self_convention)]
 pub trait CanonicalVTable<Array> {
     fn into_canonical(&self, array: Array) -> VortexResult<Canonical>;
+
+    fn canonicalize_into(&self, array: Array, _builder: &mut dyn ArrayBuilder) -> VortexResult<()> {
+        todo!(
```

Comment:
Use default and log::debug the encoding
        `builder.extend_from_array(array.into_array())`